### PR TITLE
[FIX]: Add unit tests for timed-silence expiry

### LINKED ISSUE #978: Add unit tests for timed-silence expiry (the clearExpiredSilences lazy sweep)
## Problem
Silencing an alert "until <time>" is undone lazily: every alert listing calls `clearExpiredSilences` (`apps/server/src/dal/alertRepository.ts`) via `expireSilences` in `alert.bl.ts`, which un-silences alerts whose `silenced_until` has passed and writes a history entry. Existing tests only ever silence with a far-future timestamp, so the expiry path has zero coverage.

## What to test
New file `apps/server/tests/silenceExpiry.test.ts` (use `setupDB`, `setupExpressApp`, `setupUserWithToken` from `tests/setup.ts`; seed alerts with `db.prepare(...)` as `tests/rootCause.test.ts` does):
- an alert silenced with `silencedUntil` in the past reads as `isSilenced: false` on the next `GET /api/v1/alerts`
- a future `silencedUntil` stays silenced
- `silencedUntil: null` (indefinite) never expires
- the swept alert gets an `UNSILENCED` history event with description "Silence expired" (`GET /alerts/:id/history`)
- the repository method returns the swept ids, and a second call returns `[]` (idempotent)
- boundary: `silenced_until == now` expires (the comparison is `<=`)

*(easy — `ALERTS_SNAPSHOT_TTL_MS=0` is already set by `setup.ts`, so seeded rows are visible on the next request)*

### PR DESCRIPTION
**Pull request by mateuscmtropical**
## Issue Reference

Closes #978

---

## What Was Changed

New `apps/server/tests/silenceExpiry.test.ts`, covering the timed-silence expiry path:

- a silence whose `silencedUntil` has passed reads as `isSilenced: false` on the next `GET /api/v1/alerts`
- a future `silencedUntil` stays silenced
- `silencedUntil: null` never expires
- the swept alert gets an `UNSILENCED` "Silence expired" history event
- `clearExpiredSilences` returns the swept ids, and a second call returns `[]`
- `silenced_until == now` expires (the comparison is `<=`)

---

## Why Was It Changed

The expiry path had no test coverage; existing tests only silence with a far-future timestamp.

---

## Screenshots

N/A (tests only)

---

## Additional Context (Optional)

`pnpm --filter @OpsiMate/server test` passes (30 files, 416 tests), and prettier passes. Each case fails if the matching behaviour is broken (`<=` changed to `<`, the history description changed, the sweep removed from the listing, or `NULL` included in the sweep).

## Summary by CodeRabbit

- **Tests**
  - Added coverage for time-based alert silence expiry.
  - Verified alerts with expired silences are reported as unsilenced, while future and indefinite silences remain active.
  - Confirmed expiry history events are recorded and expired silences are cleaned up correctly.
  - Covered the boundary case where a silence expires exactly at the current time, ensuring it is treated as expired.
  - Verified repeated cleanup does not return already-cleared silence IDs.

### CODE REVIEW

File: apps/server/tests/silenceExpiry.test.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+interface ListingBody {
+	data: { alerts: ListedAlert[] };
+}
+
+interface HistoryBody {
+	data: { data: HistoryEntry[] };
+}
+
+const insertSilencedAlert = (id: string, silencedUntil: string | null) => {
+	const now = new Date().toISOString();
```
[the remark this answers is not in the document]
Comment (mateuscmtropical) [reply apps/server/tests/silenceExpiry.test.ts:44]: Kept the real clock here. The boundary test captures `nowIso` once and uses that same string for both `silenced_until` and `clearExpiredSilences(nowIso)`, so the two instants can't differ. The other cases sit an hour away from now. Faking `Date` would also put the fixed instant before the JWT that `setupUserWithToken` issues in `beforeAll`. The nitpick about nested inline types is fixed in the latest commit.

**Comment by idanlodzki** [comment]
Verified on top of main (6/6, Prettier, lint clean at the strict setting) and CI is green. This hits every bullet in #978 — past/future/indefinite, the "Silence expired" history event, the idempotent sweep and the inclusive boundary — and the named interfaces for the response shapes are exactly the repo convention. I pushed one cosmetic commit (extensionless import, like the other server tests) rather than bounce it. Merging — thanks, and welcome! If you like the project, a ⭐ on the repo helps us a lot.
